File: client/src/app/shared/models/electricBikeProduct.ts
export enum ElectricBikeCategory {
  ElectricBikeModel = 1,
  ElectricBikePart = 2,
}

export interface ElectricBikeProduct {
  id: string;
  name: string;
  brand: string;
  brandName: string;
  model: string;
  category: ElectricBikeCategory;
  categoryName: string;
  description: string;
  price: number;
  stockQuantity: number;
  pictureUrl: string;
  voltage: string | null;
  power: string | null;
  batteryCapacity: string | null;
  compatibility: string | null;
  companyId: string;
  companyName: string;
  brandId: string;
  createdAt: Date;
  updatedAt: Date;
  metadata: Record<string, string>;
  isUsed?: boolean;
}

export interface CreateElectricBikeProduct {
  name: string;
  brand: string;
  model: string;
  category: ElectricBikeCategory;
  description: string;
  price: number;
  stockQuantity: number;
  pictureUrl: string;
  voltage: string | null;
  power: string | null;
  batteryCapacity: string | null;
  compatibility: string | null;
  companyId: string;
  brandId: string;
  metadata?: Record<string, string>;
  isUsed?: boolean;
}

export interface UpdateElectricBikeProduct {
  id: string;
  name: string;
  brand: string;
  model: string;
  category: ElectricBikeCategory;
  description: string;
  price: number;
  stockQuantity: number;
  pictureUrl: string;
  voltage: string | null;
  power: string | null;
  batteryCapacity: string | null;
  compatibility: string | null;
  companyId: string;
  brandId: string;
  metadata?: Record<string, string>;
  isUsed?: boolean;
}

File: Core/Entities/ElectricBikeProduct.cs
using System;
using System.Collections.Generic;

namespace Core.Entities
{
    public enum ElectricBikeCategory
    {
        ElectricBikeModel = 1,
        ElectricBikePart = 2
    }

    public class ElectricBikeProduct : BaseEntity
    {
        public string Name { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public ElectricBikeCategory Category { get; set; }
        public string Description { get; set; }

        public decimal Price { get; set; }
        public int StockQuantity { get; set; }
        public string PictureUrl { get; set; }

        public string? Voltage { get; set; }
        public string? Power { get; set; }
        public string? BatteryCapacity { get; set; }
        public string? Compatibility { get; set; }

        public int CompanyId { get; set; }
        public Company Company { get; set; }

        public int BrandId { get; set; }
        public Brand BrandEntity { get; set; }

        public Dictionary<string, string> Metadata { get; set; } = new Dictionary<string, string>();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
